assets/js/components/ModulesList.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * ModulesList component.
+ *
+ * Site Kit by Google, Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	showErrorNotification,
+} from '../util';
+import Data from 'googlesitekit-data';
+import Link from './Link';
+import ModuleIcon from './ModuleIcon';
+import GenericError from './notifications/generic-error';
+import ModuleSettingsWarning from './notifications/module-settings-warning';
+import VisuallyHidden from './VisuallyHidden';
+import { STORE_NAME as CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+
+const { useSelect, useDispatch } = Data;
+
+function ModulesList( { moduleSlugs } ) {
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const modulesData = useSelect( ( select ) => select( CORE_MODULES ).getModules() ?? {} );
+
+	/**
+	 * Handle setup module click event.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param {string} slug Module slug.
+	 */
+	const handleSetupModule = useCallback( async ( slug ) => {
+		try {
+			const { response } = await activateModule( slug );
+
+			// Redirect to ReAuthentication URL
+			global.location.assign( response.moduleReauthURL );
+		} catch ( err ) {
+			showErrorNotification( GenericError, {
+				id: 'setup-module-error',
+				title: __( 'Internal Server Error', 'google-site-kit' ),
+				description: err.message,
+				format: 'small',
+				type: 'win-error',
+			} );
+		}
+	}, [ activateModule ] );
+
+	// Filter specific modules
+	const moduleObjects = Array.isArray( moduleSlugs ) && moduleSlugs.length
+		? moduleSlugs
+			.filter( ( slug ) => modulesData.hasOwnProperty( slug ) )
+			.reduce( ( acc, slug ) => ( { ...acc, [ slug ]: modulesData[ slug ] } ), {} )
+		: modulesData;
+
+	// Filter out internal modules.
+	const modules = Object.values( moduleObjects ).filter( ( module ) => ! module.internal );
+
+	// Map of slug => name for every module that is active and completely set up.
+	const completedModuleNames = modules
+		.filter( ( module ) => module.active && module.setupComplete )
```

Comment:
Modules, received from the `core/modules` datastore, don't have `setupComplete` property. Use `connected` instead.